ssl/quic/quic_rx_frame.c

```diff
@@ -0,0 +1,391 @@
+            uint64_t max_stream_data;
+
+            if (!ossl_quic_wire_decode_frame_data_blocked(pkt, &stream_id,
+                                                          &max_stream_data))
+                return 0;
+            /* ADD CODE to send |max_stream_data| to the connection manager */
+            break;
+        }
+        case OSSL_QUIC_FRAME_TYPE_NEW_CONN_ID: {
+            OSSL_QUIC_FRAME_NEW_CONN_ID frame_data;
+
+            if (!ossl_quic_wire_decode_frame_new_conn_id(pkt, &frame_data))
+                return 0;
+            /* ADD CODE to send |frame_data.data| to the connection manager */
+            break;
+        }
+        case OSSL_QUIC_FRAME_TYPE_RETIRE_CONN_ID: {
+            OSSL_QUIC_FRAME_RETIRE_CONN_ID frame_data;
+
+            if (!ossl_quic_wire_decode_frame_retire_conn_id(pkt, &frame_data))
+                return 0;
+            /* ADD CODE to send |frame_data.data| to the connection manager */
+            break;
+        }
+        case OSSL_QUIC_FRAME_TYPE_PATH_CHALLENGE: {
+            OSSL_QUIC_FRAME_PATH_CHALLENGE frame_data;
+
+            if (!ossl_quic_wire_decode_frame_PATH_CHALLENGE(pkt, &frame_data))
+                return 0;
+            /* ADD CODE to send |frame_data.data| to the connection manager */
+            break;
+        }
+        case OSSL_QUIC_FRAME_TYPE_PATH_RESPONSE: {
+            OSSL_QUIC_FRAME_PATH_RESPONSE frame_data;
+
+            if (!ossl_quic_wire_decode_frame_path_response(pkt, &frame_data))
+                return 0;
+            /* ADD CODE to send |frame_data.data| to the connection manager */
+            break;
+        }
+        case OSSL_QUIC_FRAME_TYPE_CONN_CLOSE_TRANSPORT:
+        case OSSL_QUIC_FRAME_TYPE_CONN_CLOSE_APP: {
+            OSSL_QUIC_FRAME_CONN_CLOSE frame_data;
+
+            if (!ossl_quic_wire_decode_frame_conn_close(pkt, &frame_data))
+                return 0;
+            /* ADD CODE to send |frame_data.data| to the connection manager */
+            break;
+        }
+        case OSSL_QUIC_FRAME_TYPE_HANDSHAKE_DONE: {
+            OSSL_QUIC_FRAME_HANDSHAKE_DONE frame_data;
+
+            if (!ossl_quic_wire_decode_frame_handshake_done(pkt, &frame_data))
+                return 0;
+            /* ADD CODE to send |frame_data.data| to the handshake manager */
+            break;
+        }
+        }
+    }
+
+    return 1;
+}
+
+static int ossl_quic_handle_frames(OSSL_QRL_RX_PKT *qpacket)
+{
+    const void *packet_payload
+        = qpacket->packet_data + qpacket->packet_header_length;
+    size_t packet_payload_len
+        = qpacket->packet_data_length - qpacket->packet_payload_offset;
+    PACKET pkt;
+    OSSL_ACKM_RX_PKT ackm_data;
+    uint64_t frame_type;
+    int ok = 0;                  /* Assume the worst */
+
+    /* Phase 0: special case packet types that have no payload */
+    if (qpacket->type == QUIC_V1_LONG_PKT_TYPE_RETRY) {
+        /* ADD CODE to handle a retry */
+        goto success;
```

Comment:
Probably should confirm that `packet_payload_len == 0` in this case?

Reply:
That had me look...

Hummmmmmmmmmm

There seems to be a bug in PR #18949, see this comment: https://github.com/openssl/openssl/pull/18949/files#r945600155